src/lib.rs

```diff
@@ -702,8 +781,11 @@ impl<A: Arbitrary + Ord> Arbitrary for BinaryHeap<A> {
         u.arbitrary_take_rest_iter()?.collect()
     }
 
-    fn size_hint() -> (usize, Option<usize>) {
-        crate::size_hint::and(<usize as Arbitrary>::size_hint(), (0, None))
+    #[inline]
+    fn size_hint(depth: usize) -> (usize, Option<usize>) {
+        crate::size_hint::recursion_guard(depth, |depth| {
+            crate::size_hint::and(<usize as Arbitrary>::size_hint(depth), (0, None))
```

Comment:
I don't see why we need it for the collections, we're not calling `A::size_hint` anywhere since we just lazily provide a minimum bound of `0` (plus `usize`)

Reply:
good catch!